libxml2 2.14 made xmlBuffer.content and xmlParserCtxt.lastError
deprecated, and the ATTRIBUTE_UNUSED catchXMLError signature no longer
parses as an errorSAXFunc.

--- a/osinfo/osinfo_loader.c	2024-10-29 18:28:59.000000000 +0100
+++ b/osinfo/osinfo_loader.c	2026-10-18 04:21:54.619830255 +0200
@@ -354,7 +354,7 @@
         xmlXPathFreeObject(obj);
         OSINFO_LOADER_SET_ERROR(err, "Cannot format stylesheet");
     }
-    ret = g_strdup((char *)buf->content);
+    ret = g_strdup((char *)xmlBufferContent(buf));
 
     xmlBufferFree(buf);
     xmlXPathFreeObject(obj);
@@ -1902,28 +1902,31 @@
 }
 
 static void
-catchXMLError(void *ctx, const char *msg ATTRIBUTE_UNUSED, ...)
+catchXMLError(void *ctx, const char *msg, ...)
 {
     xmlParserCtxtPtr ctxt = (xmlParserCtxtPtr) ctx;
+    const xmlError *xmlErr;
+    gchar *xmlmsg;
 
-    if (ctxt && ctxt->_private) {
-        GError **err = ctxt->_private;
-        if (!error_is_set(err)) {
-            gchar *xmlmsg;
-            if (ctxt->lastError.file) {
-                xmlmsg = g_strdup_printf("%s:%d: %s",
-                                         ctxt->lastError.file,
-                                         ctxt->lastError.line,
-                                         ctxt->lastError.message);
-            } else {
-                xmlmsg = g_strdup_printf("at line %d: %s",
-                                         ctxt->lastError.line,
-                                         ctxt->lastError.message);
-            }
-            OSINFO_LOADER_SET_ERROR(ctxt->_private, xmlmsg);
-            g_free(xmlmsg);
-        }
+    if (!ctxt || !ctxt->_private)
+        return;
+    if (error_is_set(ctxt->_private))
+        return;
+    if (!(xmlErr = xmlCtxtGetLastError(ctx)))
+        return;
+
+    if (xmlErr->file) {
+        xmlmsg = g_strdup_printf("%s:%d: %s",
+                                 xmlErr->file,
+                                 xmlErr->line,
+                                 xmlErr->message);
+    } else {
+        xmlmsg = g_strdup_printf("at line %d: %s",
+                                 xmlErr->line,
+                                 xmlErr->message);
     }
+    OSINFO_LOADER_SET_ERROR(ctxt->_private, xmlmsg);
+    g_free(xmlmsg);
 }
 
 static void osinfo_loader_process_xml(OsinfoLoader *loader,
